### PR TITLE
Add `link-to-file-in-file-history` feature

### LINKED ISSUE #1979: In file History pages, add links to the file itself
Example: https://github.com/sindresorhus/refined-github/commits/2190e1a6f68e483f43bfda3a03d82542cda6f05f/source/features/show-names.js

There's a `<>` icon that points to the whole repo at the time of the commit, but this lacks a link to the file itself.

I think GitHub doesn't have a link here because they reused a regular "commits list", so the link to the whole repo makes sense. e.g. https://github.com/sindresorhus/refined-github/commits/master

But if a file is selected, then this fundamental link is missing.

### PR DESCRIPTION
**Pull request by HardikModha**
Closes #1979 

The feature can be tested on
- [Refined GitHub Readme](https://github.com/sindresorhus/refined-github/commits/master/readme.md)
- [React Readme](https://github.com/facebook/react/commits/master/README.md) - Commits with verified badge and comments
- [Spring Project File](https://github.com/spring-projects/spring-framework/commits/master/spring-core/src/main/java/org/springframework/core/ConfigurableObjectInputStream.java) - Very long path
- [Spring Project Directory](https://github.com/spring-projects/spring-framework/commits/master/spring-core/src/main/java/org/springframework/core/serializer) - Directory

### CODE REVIEW

File: source/libs/page-detect.ts [hunk 1]
```diff
@@ -97,6 +97,16 @@ export const isTrending = (): boolean => location.pathname === '/trending' || lo
 
 export const isUserProfile = (): boolean => select.exists('.user-profile-nav');
 
+export const isHistoryPage = (): boolean => {
```
Comment (bfred-it) [inline source/libs/page-detect.ts]: These detections are ideally URL-based. It’s best you use `isCommitList` and then detect the required elements in the feature itself.

File: source/features/link-to-file-in-file-history.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import select from 'select-dom';
+import features from '../libs/features';
+import {getOwnerAndRepo} from '../libs/utils';
+import {file} from '../libs/icons';
+import {groupButtons} from '../libs/group-buttons';
+
+function init(): void {
+	const {repoName} = getOwnerAndRepo();
+
+	const pathToFile = select('.breadcrumb')!.textContent!.trim().replace(new RegExp(`^History for ${repoName}`), '');
```
Comment (bfred-it) [inline source/features/link-to-file-in-file-history.tsx]: Find the `path` from `getRepoPath()`

File: source/features/link-to-file-in-file-history.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+	for (const commitLinkCell of select.all<HTMLDivElement>('.commit-links-cell')) {
+		const fileAnchor = (
+			<a
+				href={`${select<HTMLAnchorElement>('.commit-links-group a.btn', commitLinkCell)!.pathname.replace('/commit/', '/blob/')}${pathToFile}`}
```
Comment (bfred-it) [inline source/features/link-to-file-in-file-history.tsx]: You can move this to a variable

File: source/features/link-to-file-in-file-history.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import {getOwnerAndRepo} from '../libs/utils';
+import {file} from '../libs/icons';
+import {groupButtons} from '../libs/group-buttons';
+
+function init(): void {
+	const {repoName} = getOwnerAndRepo();
+
+	const pathToFile = select('.breadcrumb')!.textContent!.trim().replace(new RegExp(`^History for ${repoName}`), '');
+
+	for (const commitLinkCell of select.all<HTMLDivElement>('.commit-links-cell')) {
```
Comment (bfred-it) [inline source/features/link-to-file-in-file-history.tsx]: What if you select the `tree` link that’s going to be next to this button? And then use `.closest` to get this cell. I think this enables some optimizations

File: source/features/link-to-file-in-file-history.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
+/**
+This feature adds direct link to file/directory when viewing the history.
+GitHub provides a link but it points to the root of the repo which is not that helpful.
```
Comment (bfred-it) [inline source/features/link-to-file-in-file-history.tsx]: You can drop “this feature” and the second line

File: source/libs/page-detect.ts (hunk 1 above)
Comment (HardikModha) [reply source/libs/page-detect.ts]: Sure. I will. Maybe we can move it back to `page-detect` if we have another feature for the history pages.

Comment (HardikModha) [reply source/libs/page-detect.ts]: No? 😅 No worries. We won't :)

File: source/features/link-to-file-in-file-history.tsx (hunk 4 above)
Comment (HardikModha) [reply source/features/link-to-file-in-file-history.tsx]: What kind of optimizations? Can you please tell me more? I selected the parent first and then applied the `tree` link selectors to reduces the chances of any false matches.

Comment (bfred-it) [reply source/features/link-to-file-in-file-history.tsx]: fe9629b

I always try to reuse elements as much as possible, for example in this case I just got one element to get:

- insertion position
- file path (which it turns out isn't needed)
- starting point to add `width: auto`

Comment (HardikModha) [reply source/features/link-to-file-in-file-history.tsx]: Simply great! 👍 I wasn't aware of this automatic redirection thing. 😄 TIL :)

**Comment by bfred-it** [comment]
2000 ✨

**Pull request merged**